radix-engine/src/system/system_callback.rs

```diff
@@ -1582,27 +1588,24 @@ impl<V: SystemCallbackObject> KernelTransactionExecutor for System<V> {
                 }
             }
             .and_then(|_| {
-                match hash_nullification {
+                if match hash_nullification {
```

Comment:
NITPICK/OPINION/OPTIONAL: I find `if match` a little hard to read - perhaps having a separate named boolean variable `let charge_for_nullification_check = match ...;` might be a little easier?